pkg/api/vlabs/types.go

```diff
@@ -118,6 +118,14 @@ const (
 // OrchestratorProfile contains Orchestrator properties
 type OrchestratorProfile struct {
 	OrchestratorType OrchestratorType `json:"orchestratorType"`
+	KubernetesConfig KubernetesConfig `json:"kubernetesConfig,omitempty"`
+}
+
+// KubernetesConfig contains the Kubernetes config structure, containing
+// Kubernetes specific configuration
+type KubernetesConfig struct {
```

Comment:
I'm not 100% tracking what each option looks like and that json is not valid, it would need to be an object of key/value pairs, not an array.

I like the wiring up of the internal model's KubernetesConfig to the one in vlabs to expose these options. I think this is exactly what Onur is doing for letting users enable CNI integration for the Kubernetes orchestrator.

I think it's okay to duplicate fields in various Kube/DCOS/SwarmConfig objects... otherwise we will end up with this giant list of options, some of which only apply to certain orchestrators and requires us to write a bunch of validation code for which permutations of flags are allowed based on the selected orchestrator.